presto-spi/src/main/java/com/facebook/presto/spi/page/PageDataOutput.java

```diff
@@ -34,7 +35,7 @@ public PageDataOutput(SerializedPage serializedPage)
     @Override
     public long size()
     {
-        return SIZE_OF_INT * 3 + SIZE_OF_BYTE + serializedPage.getSizeInBytes();
+        return SIZE_OF_INT * 3 + SIZE_OF_BYTE + SIZE_OF_LONG + serializedPage.getSizeInBytes();
```

Comment:
can we use a constant for the metadata size to PAGE_METADATA_SIZE, which is also in SerializedPageWriteListener

Reply:
Good point. I had to move the constant to the `PagesSerdeUtil` though to make it available in both modules.